src/modules/url-preview/fetchFromMetaData.js

```
@@ -0,0 +1,137 @@
+			const match = body.match(getProperty(propsWithType[i], types[j]));
+
+			if (match && match.length) {
+				data['thumbnail_' + propsWithType[i]] = parseInt(getContent(match), 10);
+			}
+		}
+	}
+
+	const imageUrl = body.match(regexes.image);
+
+	if (imageUrl) {
+		data.thumbnail_url = getContent(imageUrl);
+	}
+
+	if (!data.title) {
+		const matches = body.match(regexes.title);
+
+		if (matches && matches.length) {
+			const title = matches[0].match(/[>][^<]*/);
+
+			if (title && title.length) {
+				data.title = decodeText(title[0].slice(1));
+			}
+		}
+	}
+
+	if (!data.description) {
+		const matches = body.match(regexes.description);
+
+		if (matches && matches.length) {
+			const description = matches[0].match(regexes.content)[0].match(/['|"][^'|^"]*/);
+
+			if (description && description.length) {
+				data.description = decodeText(description[0].slice(1));
+			}
+		}
+	}
+
+	if (Object.keys(data).length > 1) {
+		return data;
+	}
+
+	return null;
+}
+
+function extractLink(body) {
+	const res = body.match(regexes.link);
+
+	if (res && res.length) {
+		return res[0].match(/http[s]?:\/\/[^"']*/i)[0].replace(/&amp;/g, '&');
+	}
+
+	return null;
+}
+
+async function fetchData(url: string): Promise<Embed> {
+	const body = await fetch(url).then(res => res.text()).then(text => text ? text.replace(/(\r\n|\n|\r)/g, '') : '');
+	const dataUrl = extractLink(body);
+
+	let data;
+
+	if (dataUrl) {
+		data = await (await fetch(dataUrl)).json();
+	} else {
+		data = parseHTML(body);
+	}
+
+	return data;
+}
+
+export default async function (url) {
+	const contentType = await getContentType(url);
+	winston.debug(`Content-type for url: ${url}, ${contentType}`);
+	return new Promise((resolve) => {
+		if (contentType) {
+			if (contentType.indexOf('image') > -1) {
+				resolve({
+					type: 'link',
+					thumbnail_url: url,
+				});
+			} else if (contentType.indexOf('text/html') > -1) {
+				resolve(fetchData(url));
+			}
+		} else {
+			resolve(null);
```

Let's throw an error instead of resolving with null